### ISSUE TITLE
Add FlowAnomalyScorer and calibrated anomaly scoring API

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#81]
## User story

As a user applying normalizing flows to anomaly detection, I want a first-class scoring object so that I do not have to manually interpret `-flow.log_prob(x)` as an anomaly score. I want the API to make raw likelihood, typicality, latent norm, and decomposed likelihood scores explicit choices.

## Current state

`gauss_flows` currently exposes flow constructors such as `gaussianization_flow` and `coupling_gaussianization_flow`, and training utilities such as `fit_gaussianization_flow`. The standard user workflow is effectively:

```python
flow = gaussianization_flow(key, n_dims=10)
trained_flow, losses = fit_gaussianization_flow(key, flow, data)
scores = -trained_flow.log_prob(x)
```

This encourages raw negative log-likelihood as the default anomaly score.

## Target state

Add a high-level anomaly scoring API that makes the anomaly-score definition explicit and supports calibration. The API should encourage users to choose between raw NLL, typicality, latent-space scores, likelihood decomposition, and future likelihood-ratio scores.

## Proposal

Add a new module:

```text
src/gauss_flows/_src/anomaly/
    __init__.py
    scoring.py
    calibration.py
```

Expose public imports through either:

```python
from gauss_flows.anomaly import FlowAnomalyScorer
```

or from the root namespace later.

The scorer should wrap any FlowJax-like distribution with a `log_prob` method. It should not train the flow itself. It should only define and calibrate anomaly scores.

## Mathematical formulation

Given a flow density `p_theta(x)`, the naive anomaly score is:

`A_nll(x) = - log p_theta(x)`

But high likelihood is not necessarily normality. A calibrated scorer should support alternative scores, for example:

`A_typ(x) = |log p_theta(x) - μ_cal| / σ_cal`

where:

- `μ_cal = mean_{x in D_cal} log p_theta(x)`
- `σ_cal = std_{x in D_cal} log p_theta(x)`

The prediction rule is:

`is_anomaly(x) = A(x) > τ`

where `τ` is learned from calibration data, for example:

`τ = quantile({A(x_i) : x_i in D_cal}, q)`

## Proposed API

```python
from gauss_flows.anomaly import FlowAnomalyScorer, QuantileCalibrator

scorer = FlowAnomalyScorer(
    flow=trained_flow,
    score="nll",          # "nll", "typicality", "latent_norm", "base_nll", "logdet"
    reduce="none",        # "none", "mean", "sum"
)

scores = scorer.score(x)

scorer.calibrate(
    x_calib,
    calibrator=QuantileCalibrator(q=0.995),
)

labels = scorer.predict(x_test)
```

Suggested methods:

```python
class FlowAnomalyScorer:
    def __init__(self, flow, *, score="nll", reduce="none"):
        ...

    def log_prob(self, x, condition=None):
        ...

    def score(self, x, condition=None):
        ...

    def calibrate(self, x_calib, condition=None, calibrator=None):
        ...

    def predict(self, x, condition=None, threshold=None):
        ...
```

## Example use case

```python
import jax.random as jr
from gauss_flows import gaussianization_flow, fit_gaussianization_flow
from gauss_flows.anomaly import FlowAnomalyScorer, QuantileCalibrator

key = jr.key(0)
flow = gaussianization_flow(key, n_dims=x_train.shape[-1], n_layers=8)
flow, losses = fit_gaussianization_flow(key, flow, x_train)

scorer = FlowAnomalyScorer(flow, score="typicality")
scorer.calibrate(x_val, calibrator=QuantileCalibrator(q=0.995))

scores = scorer.score(x_test)
is_anomaly = scorer.predict(x_test)
```

## Tasks

- [ ] Create `src/gauss_flows/_src/anomaly/scoring.py`.
- [ ] Implement `FlowAnomalyScorer` with `score`, `log_prob`, `calibrate`, and `predict`.
- [ ] Support at least `score="nll"` and `score="typicality"`.
- [ ] Add `QuantileCalibrator` in `calibration.py`.
- [ ] Add tests for NLL scoring.
- [ ] Add tests for typicality scoring.
- [ ] Add tests for quantile calibration and prediction.
- [ ] Add public exports via `gauss_flows.anomaly`.
- [ ] Add a README or docs snippet warning that raw likelihood is not always reliable for OOD detection.
